spec.bs

```diff
@@ -713,27 +718,60 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
 
 To <dfn>score and rank a bid</dfn> given an [=auction config=] |auctionConfig|, a [=generated bid=]
 |generatedBid|, a [=leading bid info=] |leadingBidInfo|, a [=string=] |decisionLogicScript|, a
-{{unsigned long}}-or-null |biddingDataVersion|, and a [=boolean=] |isComponentAuction|:
-1. TODO: Assemble and fetch trusted scoring signals
-1. TODO: Let |adMetadata| be ... maybe include in [=generated bid=]
+{{unsigned long}}-or-null |biddingDataVersion|, a [=boolean=] |isComponentAuction|, and an [=origin=]
+|topWindowOrigin|:
+1. Let |renderURLs| be a new [=set/is empty|empty=] [=set=].
+1. Let |adComponentRenderUrls| be a new [=set/is empty|empty=] [=set=].
+1. [=set/Append=] |generatedBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=] to |renderURLs|.
+1. If |generatedBid|'s [=generated bid/ad component descriptors=] is not null:
+  1. [=set/For each=] |adComponentDescriptor| in |generatedBid|'s
+    [=generated bid/ad component descriptors=]:
+    1. [=set/Append=] |adComponentDescriptor|'s [=ad descriptor/url=] to |adComponentRenderUrls|.
+1. Let |fullSignalsUrl| be the result of [=building trusted scoring signals url=] with |auctionConfig|'s
+  [=auction config/trusted scoring signals url=], |renderURLs|, |adComponentRenderUrls|,
+  |auctionConfig|'s [=auction config/seller experiment group id=], and |topWindowOrigin|.
+
+  Note: Implementations can batch requests by collecting render URLs and ad component render URLs

```

Comment:
If this is a normative allowance it should not be in a note and should use "may"